pyaerocom/ungridded_data_metadata.py

```diff
@@ -0,0 +1,985 @@
+import abc
+import fnmatch
+import logging
+import sys
+from copy import deepcopy
+from datetime import datetime
+from typing import Any
+
+import numpy as np
+
+from pyaerocom import const
+from pyaerocom.exceptions import DataCoverageError, MetaDataError, StationNotFoundError
+from pyaerocom.helpers import isnumeric, start_stop_str
+from pyaerocom.helpers_landsea_masks import get_mask_value, load_region_mask_xr
+from pyaerocom.mathutils import in_range
+from pyaerocom.ungridded_data_container import UngriddedDataContainer
+from pyaerocom.units_helpers import get_unit_conversion_fac
+
+logger = logging.getLogger(__name__)
+
+
+if sys.version_info >= (3, 12):
+    from typing import override
+else:
+    from typing_extensions import override
+
+
+class UngriddedDataMetadata(UngriddedDataContainer):
+    """Metadata-implementation for UngriddedDataContainer implementing
+    metadata, i.e. metadata, data_revision, var_idx/variables, filter_hist and
+    _is_vertical_profile. It does not do anything on the data-structure.
+
+    This will only implement the UngriddedDataContainer partially. The data
+    part needs to be implemented independently.
+
+    """
+
+    ALLOWED_VERT_COORD_TYPES = ["altitude"]
+
+    def __init__(self):
+        self.metadata = {}
```

Comment:
Could we think of a better container than a dictionary to store metadata? What about a dataclass, or a dot-dict? OR is the benefit of the dictionary really the flexibility and mutability?

Reply:
This PR and the UngriddedDataStructured reuses the old metadata interface without any changes. And just moved the metadata-parts to ungridded_data_metadata.py . Implementing a different metadata handling is out of scope, even if the new interface opens up for that.

Reply:
Fair enough. Just something to maybe think about in future work.